### PR TITLE
[XPU] fix recompute not support bug on xpu

### PR DESCRIPTION
**Pull request by HarperCy**
### PR types
Bug fixes

### PR changes
Others

### Description
fix recompute not support bug on xpu

### CODE REVIEW

File: python/paddle/distributed/fleet/recompute/recompute.py [hunk 1]
```diff
@@ -224,6 +224,8 @@ def _recompute_without_reentrant(
         cur_device = paddle.get_device()
         if 'gpu:' in cur_device:
             fw_cuda_rng_state = paddle.get_cuda_rng_state()
+        elif 'xpu:' in cur_device:
+            fw_cuda_rng_state = paddle.get_rng_state()
```
Comment (XiaociZhang) [inline python/paddle/distributed/fleet/recompute/recompute.py:228]: 需要像下面custom device一样加上`cur_device`参数么

Comment (HarperCy) [reply python/paddle/distributed/fleet/recompute/recompute.py:228]: 试了，不需要

**Review by XiaociZhang — COMMENTED** [review]
lgtm

File: python/paddle/distributed/fleet/recompute/recompute.py (hunk 1 above)
Comment (HarperCy) [reply python/paddle/distributed/fleet/recompute/recompute.py:228]: > 

get_rng_state()不传参则里面自动获取device信息

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**